packages/backend/src/app.ts

```diff
@@ -40,16 +41,43 @@ export default function makeApp() {
     const frontendPath = '../../../../packages/frontend/build/';
 
     const path = require('path');
-    app.use(express.static(path.join(__dirname, frontendPath)));
     app.use(express.json());
     //Routes
     app.use('/API',authController.getUser, electionRouter)
     // app.use('/debug',debugRouter)
 
     app.post('/API/Token', asyncHandler(getUserToken));
 
+    // NOTE: I've removed express.static because it doesn't allow me to inject meta tags
+    // https://stackoverflow.com/questions/51120214/how-to-modify-static-file-content-with-express-static
     app.get('*', (req, res) => {
-        res.sendFile(path.join(__dirname, frontendPath + "index.html"));
+        const fs = require('fs');
+        fs.readFile(path.join(__dirname, frontendPath, req.url.split('?')[0]), 'utf8', (err:any, htmlData:string) => {
+            if(err){
+                // if the request wants a webpage, then return index.html and inject meta tags
+
+                // https://blog.logrocket.com/adding-dynamic-meta-tags-react-app-without-ssr/
+                fs.readFile(path.join(__dirname, frontendPath, 'index.html'), 'utf8', async (err:any, htmlData:string) => {
+                    if(err){
+                        console.error('Error during file reading', err);
+                        return res.status(404).end();
+                    }
+
+                    // inject tags
+                    const tags = await getMetaTags(req);
+
+                    Object.entries(tags).forEach(([key, value]) => {
+                        htmlData = htmlData.replaceAll(key, value);
```

Comment:
Any sense of how much of a performance hit this is? If there's only one of each key it might be faster to use the replace method instead since the meta tags should be towards the front if the file.

Reply:
I had that question as well, but couldn't find a clear answer if replaceAll had significantly worse performance than replace

I used replace all since some of them do come up twice, but luckily the compiled index.html isn't super long anyway